@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -80,6 +80,32 @@ export class TextElementState {
         return false;
     }
 
+    /**
+     * @returns `true` if any component of the element is really visible, `false` otherwise.
+     */
+    get reallyVisible(): boolean {
+        if (this.m_textRenderState !== undefined && this.m_textRenderState.isReallyVisible()) {
+            return true;
+        }
+
+        const iconRenderState = this.iconRenderState;
+        if (iconRenderState !== undefined && iconRenderState.isReallyVisible()) {
+            return true;
+        }
+
+        const iconRenderStates = this.iconRenderStates;
+        if (iconRenderStates === undefined) {
+            return false;
+        }
+
+        for (const state of iconRenderStates) {
+            if (state.isReallyVisible()) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
I'd rather avoid having the almost exact copy of isVisible(), better change isVisible behaviour so that it checks opacity.